src/NServiceBus.Core/Timeout/Hosting/Windows/TimeoutDispatcherProcessor.cs

```diff
@@ -32,7 +32,16 @@ public bool Handle(TransportMessage message)
 
             if (TimeoutsPersister.TryRemove(timeoutId, out timeoutData))
             {
-                MessageSender.Send(timeoutData.ToTransportMessage(), timeoutData.ToSendOptions(Configure.LocalAddress));
+                try
+                {
+                    MessageSender.Send(timeoutData.ToTransportMessage(), timeoutData.ToSendOptions(Configure.LocalAddress));
+                }
+                catch
+                {
+                    timeoutData.Id = null;
+                    timeoutData.Time = DateTime.UtcNow.AddSeconds(5);
+                    TimeoutsPersister.Add(timeoutData);
```

Comment:
Technically this might still loose data since if the `.Add` blows up but this does make it less likely to happen.

But there is a chance that both the queuing system and the storage goes bad at the same time.  @Particular/nservicebus what do you all think?